### PR TITLE
Landing Page

### PR DESCRIPTION
**Pull request by anushkumar27**
# Proposed Changes

Implementation of Landing Page View (view only), its implemented as a stateless widget the functionality is added post completing all the relevant views.

|Figma Design| Flutter Implementation|
|---|---|
|image|image|

# Jira Stories

- [LAB-315](https://oregonstate-innovationlab.atlassian.net/browse/LAB-315)

# Checklist

- [x] This has been tested
- [x] Self-documenting code
- [x] Code is linted

# Instructions to review
Code inspection and feedback

### CODE REVIEW

File: lib/screens/home_screen.dart [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+class HomeScreen extends StatefulWidget {
+  const HomeScreen({super.key});
+
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  final double _initFabHeight = 100.0;
+  double _fabHeight = 0;
```
Comment (wbraynen) [inline lib/screens/home_screen.dart]: What is "fab"?

Comment (anushkumar27) [reply lib/screens/home_screen.dart]: Floating Action Button (FAB) - In this case, it's the "Contribute +" button that needs to be moved up and down based on the sliding bottom panel.

File: lib/screens/home_screen.dart [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+              onMapCreated: _onMapCreated,
+              initialCameraPosition: CameraPosition(
+                target: _center,
+                zoom: 11.0,
+              ),
+            ),
+            onPanelSlide: (double pos) => setState(() {
+                  _fabHeight = pos * (_panelHeightOpen - _panelHeightClosed) +
+                      _initFabHeight;
+                })),
```
Comment (wbraynen) [inline lib/screens/home_screen.dart]: Would it be better to make it self-documenting?  For example:
```
 onPanelSlide: (double position) => setState(() {
        updateContributeButton(position)
      })),
```

File: lib/screens/home_screen.dart [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                suffixIcon: Icon(Icons.cancel_outlined),
+                filled: true,
+                fillColor: Colors.white,
+                border: OutlineInputBorder(),
+                labelText: 'Search',
+              )),
+        )
+      ],
+    );
+  }
```
Comment (wbraynen) [inline lib/screens/home_screen.dart:112]: Maybe more self documenting?  Would it be clearer if we wrote something like the following instead?

```
  @override
  Widget build(BuildContext context) {
    _explorationPanelHeightOpen = MediaQuery.of(context).size.height * .50;

    let explorationPanel = SlidingUpPanel(
        maxHeight: _explorationPanelHeightOpen,
        minHeight: _explorationPanelHeightClosed,
        parallaxEnabled: true,
        parallaxOffset: .5,
        renderPanelSheet: false,
        panel: _floatingPanel(context),
        collapsed: _floatingCollapsed(context),
        body: GoogleMap(
          onMapCreated: _onMapCreated,
          initialCameraPosition: CameraPosition(
            target: _center,
            zoom: 11.0,
          ),
        ),
        onPanelSlide: (double position) => setState(() {
              udpateContributeButtonPosition(position)
        })
    )

    let contributeButton = Positioned(
      right: 20.0,
      bottom: _fabHeight,
      child: FloatingActionButton.extended(
        onPressed: () {
          // Add your onPressed code here!
        },
        label: const Text('Contribute'),
        icon: const Icon(Icons.add),
      )
    )

    let searchBox = Container(
        margin: const EdgeInsets.fromLTRB(20.0, 20.0, 20.0, 20.0),
        child: const TextField(
            autofocus: false,
            decoration: InputDecoration(
              hintText: "Name, Category, Food trucks..",
              prefixIcon: Icon(Icons.search),
              suffixIcon: Icon(Icons.cancel_outlined),
              filled: true,
              fillColor: Colors.white,
              border: OutlineInputBorder(),
              labelText: 'Search',
        ))
    )

    return Stack(
      alignment: Alignment.topCenter,
      children: <Widget>[
        explorationPanel,
        contributeButton,
        searchBox
      ]
    );
  }
```

File: lib/screens/home_screen.dart [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+              ),
+            ],
+          ),
+          const SizedBox(
+            height: 18.0,
+          ),
+          for (int i = 0; i < 6; i++) ...[
+            const HorizontalCard()
+          ],
+        ],
```
Comment (wbraynen) [inline lib/screens/home_screen.dart]: Clean up the same way, so it's not so mysterious and intimidating (wall of text and all)?

File: lib/widgets/horizontal_card.dart [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        shape: RoundedRectangleBorder(
+          side: BorderSide(
+            color: Theme.of(context).colorScheme.outline,
+          ),
+          borderRadius: const BorderRadius.all(Radius.circular(12)),
+        ),
+        child: Container(
+          padding: const EdgeInsets.all(20),
+          child: Row(
+            children: [
```
Comment (wbraynen) [inline lib/widgets/horizontal_card.dart]: More children!  Name your children!  Here and elsewhere.

**Review by wbraynen — CHANGES_REQUESTED** [review]
Chatted with @anushkumar27 and we decided he'll try self-documenting code and see how it feels.  If it doesn't make things better, then never mind :).

Self-documenting code:
https://stackoverflow.com/questions/209015/what-is-self-documenting-code-and-can-it-replace-well-documented-code
See the answer with 407 upvotes

Not sure this is an example you should be following as role model, but if you are curious, there is also this sort of production code:
https://github.com/google/tcmalloc

**Review by wbraynen — APPROVED** [review]
🎉

**Comment by wbraynen** [comment]
I do recommend pressing "squash and merge" for this when you merge this PR.

**Pull request merged**